### ISSUE TITLE
HA Deployment: Scheduled tasks shall by synchronised between multiple services

### ISSUE THREAD

**Issue by mzueva** [epam/cloud-pipeline#688]
Related to #292

# Background
We need to run a multiple copies of our services (API, notifier, elasticsearch synchronizer, etc.) to support HA deployment, but most of the scheduled tasks, e.g. autoscaling in API, cannot run in parallel and we shall allow only one running copy of such task at once among the whole deployment.

In order to implement such synchronisation we'll consider that only one instance of a service is an active master that is allowed to run scheduled tasks, all other instances shall be passive and skip scheduled tasks. In order to determine whether current service instance is an active master we'll add a special endpoint (subject of a separate task/issue) that will return name of current master, so any service will be able to determine whether its host is master or not.

Also to prevent some race condition and failure situations each scheduled task shall acquire a distributed lock in order to run.

# Approach

- Introduce an  `Around` aspect for scheduled tasks (**TBD** check that aspect will be actually invoked for scheduled tasks). This aspect shall make a http GET request to some endpoint (configurable via `SystemPreference` or application.properties) and check if returned master name matches current host name. If current service is a master, scheduled task shall run, otherwise it is skipped.
- Add `ShedLock` library to the related service and use `@SchedulerLock` for Scheduled methods. **TBD** Check that this works for API scheduling approach (with `AbstractSchedulingManager`)

**Comment by mzueva** [comment]
@Wedds configure `ShedLock` annotation for es module `ElasticsearchAgentService` class 
Docs: https://www.baeldung.com/shedlock-spring
